app/Http/Controllers/API/AuthController.php

```diff
@@ -67,12 +71,45 @@ public function register(RegisterRequest $request, UserService $userService)
         $user = $userService->create($request->all());
         if($request->input("type") === "charity") {
             $charity = (new CharityService)->create([
```

Comment:
Salaam @afmire877 ,

You'll need to implement the saving of the charity and collection records in a database transaction here. If there is a validation failure in either model then we don't want to save anything to the database. 

You can see how we did it previously in App\Services\User::create()